shutter.py

```diff
@@ -0,0 +1,196 @@
+
+        self.char_rotation_speed = serv_fan.configure_char(
+            'RotationSpeed', setter_callback=self.set_rotation_speed)
+        self.char_rotation_direction = serv_fan.configure_char(
+            'RotationDirection', setter_callback=self.set_rotation_direction)
+
+    def set_rotation_speed(self, value):
+        logging.debug("Rotation speed changed: %s", value)
+
+    def set_rotation_direction(self, value):
+        logging.debug("Rotation direction changed: %s", value)
+
+class LightBulb(Accessory):
+    """Fake lightbulb, logs what the client sets."""
+
+    category = CATEGORY_LIGHTBULB
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        serv_light = self.add_preload_service('Lightbulb')
+        self.char_on = serv_light.configure_char(
+            'On', setter_callback=self.set_bulb)
+
+    def set_bulb(self, value):
+        logging.info("Bulb value: %s", value)
+
+class GarageDoor(Accessory):
+    """Fake garage door."""
+
+    category = CATEGORY_GARAGE_DOOR_OPENER
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.add_preload_service('GarageDoorOpener')\
+            .configure_char(
+                'TargetDoorState', setter_callback=self.change_state)
+
+    def change_state(self, value):
+        logging.info("GDC value: %s", value)
+        self.get_service('GarageDoorOpener')\
+            .get_characteristic('CurrentDoorState')\
+            .set_value(value)
+
+
+class WindowCovering(Accessory):
+    """Fake WindowCovering."""
+
+    category = CATEGORY_WINDOW_COVERING
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        """Quick assign basic accessory information."""
+        self.set_info_service(firmware_revision=2,  manufacturer="Brand",
+                model="Shutter", serial_number="0123456789")
+        
+        # Add the WindowCovering service. Also add optional characteristics to it.
+        serv_cover = self.add_preload_service(
+            'WindowCovering', chars=['ObstructionDetected'])
+
+        self.char_target_pos = serv_cover.configure_char(
+            'TargetPosition', setter_callback=self.set_target_position)
+            
+        self.char_state = serv_cover.configure_char(
+            'PositionState', setter_callback=self.set_position_state)
+            
+        self.char_cur_pos = serv_cover.configure_char(
+            'CurrentPosition', setter_callback=self.set_current_position)
+                                     
+        #serv_cover.configure_char('PositionState', value=2) # STOPPED
+        #serv_cover.configure_char('ObstructionDetected', value=1)
+                                     
+    def set_target_position(self, value):
+        logging.info("WindowCovering TargetPosition value: %s", value)
+        self.get_service('WindowCovering')\
+            .get_characteristic('TargetPosition')\
+            .set_value(value)
```

Comment:
Don't set the char to the value that it posted. This isn't necessary. Just remove it.